app/src/main/java/com/pajato/android/gamechat/common/PlayModeManager.java

```diff
@@ -52,32 +64,63 @@
 
     // Public instance variables
 
-    /** Identifies the types of play. */
-    public enum PlayModeType {computer, local, user}
-
     // Protected instance variables
 
-    /** The current play mode popup menu for the experience being enjoyed. */
-    private PopupMenu mPlayModePopup;
-
-    // Private  instance variables
-
-    /** The listener for menu item clicks on the play mode popup menu */
-    private PlayModeClickListener mListener = new PlayModeClickListener();
+    /** The current play mode menu for the experience being enjoyed. */
+    private PopupWindow mPlayModePopupWindow;
 
     // Public instance methods.
 
+    /** Close the play-mode popup window */
+    public void closePlayModeMenu() {
+        mPlayModePopupWindow.dismiss();
+    }
+
     /** Dismiss the play mode menu if it is not null */
     public void dismissPlayModeMenu() {
-        if (mPlayModePopup != null)
-            mPlayModePopup.dismiss();
+        if (mPlayModePopupWindow != null)
+            mPlayModePopupWindow.dismiss();
+    }
+
+    /** Handle a use selection in the play mode menu */
```

Comment:
I think this method should have two parts: a setup/validation part and a processing part.  The setup validation part should be something like:

    Object payload = view.getTag();
    boolean flag = payload != null && payload instanceof PlayModeMenuEntry;
    PlayModeMenuEntry entry = flag ? (PlayModeMenuEntry) payload : null;
    Account member =  entry != null ? MM.instance.getMember(entry.groupKey, entry.accountKey) : null;
    if (member == null) 
        return;
The rest of the code might be compacted a bit as I've indicated in subsequent comments.